usr/share/rear/lib/network-functions.sh

```diff
@@ -723,3 +723,28 @@ is_bonding_device ()
    [ -f "/sys/class/net/$1/bonding/slaves" ]
 }
 
+#  ^(([1-9]|[12]?[0-9]{1,2}|2([0-4][0-9]|5[0-5]))\.){3}([1-9]|[12]?[0-9]{1,2}|2([0-4][0-9]|5[0-5]))$
+
+# Retun 0 if args is a valid IPV4 ip_address
+function is_ip() {
+    local test_ip=$1
+    [ -z $test_ip ] && BugError "function is_ip() called without argument."
+
+    # test if $test_ip is a valide IPV4 address: "[1 to 255].[1 to 255].[1 to 255].[1 to 255]"
+    if [[ $test_ip =~ ^(([0-9]{1,2}|1[0-9]{2}|2([0-4][0-9]|5[0-5]))\.){3}([0-9]{1,2}|1[0-9]{2}|2([0-4][0-9]|5[0-5]))$ ]] ; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# function which get the ipv4 address from a fqdn
+function get_ip_from_fqdn()
+{
+    local fqdn=$1
+    [ -z $fqdn ] && BugError "function get_ip_from_name() called without argument."
+
+    # Retuning the 1 Field of the First line of the command `getent ahostsv4 $fqdn`
+    # if the first field is an IPv4 valid address "[1 to 255].[1 to 255].[1 to 255].[1 to 255]"
+    echo $(getent ahostsv4 $fqdn | awk '$1~/^(([0-9]{1,2}|1[0-9]{2}|2([0-4][0-9]|5[0-5]))\.){3}([0-9]{1,2}|1[0-9]{2}|2([0-4][0-9]|5[0-5]))$/ { print $1 ; exit }' )
```

Comment:
Those long regexes are very difficult to read and error prone. Also, you already have a function that checks if a strign in an IP. It is also important to check the return code of getent.

Maybe simpler like this:

```bash
ip=( $(getent ahostsv4 $fqdn) ) || Error "Could not resolve $fqdn"
is_ip "$ip" || Error "Got '$ip' from resolving $fqdn which is not an IP"
echo "$ip"
```

This uses a trick that we use a lot in ReaR. Using a Bash array like a scalar yields the first value.

Do you happen to know since when `ahostv4` is supported? Just need to make sure that it works on all *supported* ReaR distros.

Reply:
I see... But in this case, I would like to return from `getent ahostsv4 $fqdn`  : 
"the first field of (the first line where the first field is an IP)." That's why I'm using `awk` here with the same regex as `is_ip`.

 ```
getent ahostsv4 yahoo.fr
74.6.50.24      STREAM yahoo.fr
74.6.50.24      DGRAM  
74.6.50.24      RAW    
77.238.184.24   STREAM 
77.238.184.24   DGRAM  
77.238.184.24   RAW    
106.10.212.24   STREAM 
106.10.212.24   DGRAM  
106.10.212.24   RAW    
98.137.236.24   STREAM 
98.137.236.24   DGRAM  
98.137.236.24   RAW    
124.108.105.24  STREAM 
124.108.105.24  DGRAM  
124.108.105.24  RAW 
```
So, If there is several answer (like the above example) I'm sure to only get the First Line with a Real IP address (Error messages or other "junk" lines will be filtered.)

For the distro, I've tested it with sles11, sles12, ubuntu16.04, rhel7

Reply:
So far I have never seen `getent` print error messages on STDOUT. The array tricks treats a newline as a blank (it uses `$IFS` to split the input into words) so that it does exactly what you want.

Reply:
Ok ... what about this one : 
```
    # Get a list of potential IPs that resolve $fqdn
    ip=( $(getent ahostsv4 $fqdn | awk '{ print $1}') ) || Error "Could not resolve $fqdn"

    # Loop on the array still we find a valide IP.
    local i=0
    while ! is_ip ${ip[$i]} ; do
        Log "Got '${ip[$i]}' from resolving $fqdn which is not a valide IP"
        let i+=1

        # Stop looping if the next element to look at is greater that the number of element in the array.
        if test $i -gt ${#ip[@]} ; then
            LogPrintError "Can't resolve $fqdn to a valide IP address."
            return 1
        fi
    done
```